lib/spack/spack/detection/path.py

```diff
@@ -412,8 +428,14 @@ def by_path(
     """
     # TODO: Packages should be able to define both .libraries and .executables in the future
     # TODO: determine_spec_details should get all relevant libraries and executables in one call
-    executables_finder, libraries_finder = ExecutablesFinder(), LibrariesFinder()
+    executables_finder, libraries_finder = ExecutablesFinder(debug_find), LibrariesFinder(debug_find)
     detected_specs_by_package: Dict[str, Tuple[concurrent.futures.Future, ...]] = {}
+    if debug_find:
+        llnl.util.tty.info("Spack performed external detection with the following configuration:")
+        llnl.util.tty.info(f"\tMax Workers: {max_workers}")
+        llnl.util.tty.info("\tSupplied path hints:")
+        if path_hints:
+            [llnl.util.tty.info(f"\t\t{x}") for x in path_hints]
```

Comment:
I see this allows you to not use a `for` statement, but please use the `for` statement

Reply:
Fair enough, will amend.